### PR TITLE
CM-807: Upgrading build dependencies

### PR DESCRIPTION
**Pull request by SuperIzya**
All npm dependencies upgraded to remove outdated/deprecated/vulnerable ones.

Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,51 +41,49 @@
     "release:ci:minor": "release-it minor --ci",
     "release:ci:major": "release-it major --ci"
   },
-  "dependencies": {
-    "js-cookie": "^3.0.1",
-    "live-connect-common": "^1.0.0"
+  "peerDependencies": {
+    "js-cookie": "^3.0.5",
+    "live-connect-common": "file:../live-connect-common/live-connect-common-3.0.0.tgz"
```
Comment (SuperIzya) [inline package.json:46]: Will be fixed to proper dependency, when `live-connect-common` is published.

File: package.json [hunk 2]
```diff
@@ -26,65 +28,62 @@
...
     "release:ci:alpha->beta": "release-it --preRelease=beta --ci",
     "release:ci:beta->rc": "release-it --preRelease=rc --ci",
     "release:ci:patch": "release-it patch --ci",
     "release:ci:minor": "release-it minor --ci",
     "release:ci:major": "release-it major --ci"
   },
-  "dependencies": {
-    "js-cookie": "^3.0.1",
-    "live-connect-common": "^1.0.0"
+  "peerDependencies": {
```
Comment (3link) [inline package.json:44]: Why going with peer dependencies?

Comment (SuperIzya) [reply package.json:44]: They might be also set by `live-connect` or `live-connect-builder`.

File: tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,16 @@
 {
   "compilerOptions": {
     "allowJs": false,
-    "typeRoots": ["./node_modules/@types", "./src/typings"],
-    "types": ["mocha", "node"],
+    "typeRoots": ["./node_modules/@types"],
+    "types": ["mocha"],
```
Comment (wi101) [inline tsconfig.json:5]:
```suggestion
    "types": ["node"],
```
?

File: tsconfig.json [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
     "esModuleInterop": true,
     "resolveJsonModule": true,
-    "module": "CommonJS",
+    "module": "ESNext",
```
Comment (wi101) [inline tsconfig.json:8]: should we set the target as well?

Comment (SuperIzya) [reply tsconfig.json:8]: Not really. The code will be plyfilled later in `live-connect`.

File: package.json (hunk 2 above)
Comment (SuperIzya) [reply package.json:44]: Nope. You are right @mschuwalow

Comment (SuperIzya) [reply package.json:44]: But I will investigate it more closely

**Review by 3link — APPROVED** [review]

**Review by peixunzhang — APPROVED** [review]

**Review by wi101 — APPROVED** [review]

**Review by mschuwalow — APPROVED** [review]

**Pull request merged**